hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```
@@ -795,10 +801,10 @@ public boolean mkdirs(final Path f, final FsPermission permission) throws IOExce
 
   @Override
   public synchronized void close() throws IOException {
-    if (isClosed) {
+    if (isClosed()) {
```

Should we throw an exception in case someone is trying to close non initialized or already closed file system?

---

fs.close() is supposed to be idempotent IMO

---

There is chance that more two threads can attempt to close store and client which can cause similar issue what we got now.

---

@bhattmanish98 I don't see concurrency issues because close() is synchronized, and once closed it can't be closed again. Of course, once one thread has closed it, nobody else can use the instance. fix: don't do that.

---

@steveloughran Make sense, I overlooked synchronized in the method definition. PR Looks good. Approved the changes.